src/features/search/items/address-result-item.tsx

```diff
@@ -1,23 +1,30 @@
 import React from 'react';
 import { Box, Flex } from '@stacks/ui';
 import { FoundResult } from '@common/types/search-results';
-import { useAtomValue } from 'jotai/utils';
-import { accountStxBalanceResponseState } from '@store/accounts';
 import { microToStacks, truncateMiddle } from '@common/utils';
 import { AddressLink } from '@components/links';
 import { ItemIcon } from '@components/item-icon';
 import { Caption, Title } from '@components/typography';
 import { ResultItemWrapper } from '@features/search/items/result-item-wrapper';
+import { useApi } from '@common/api/client';
+import { useQuery } from 'react-query';
+import { AddressStxBalanceResponse } from '@stacks/stacks-blockchain-api-types';
 
 interface AddressResultItemProps {
   result: FoundResult;
 }
+
 export const AddressResultItem: React.FC<AddressResultItemProps> = ({ result }) => {
-  if (!result || !result.found || result.result.entity_type !== 'standard_address') return null;
   const principal = result.result.entity_id;
-  const displayName = result.result.display_name;
+  const displayName =
+    result.result.entity_type === 'standard_address' ? result.result.display_name : '';
   const truncatedPrincipal = truncateMiddle(principal, 4);
-  const stx = useAtomValue(accountStxBalanceResponseState(principal));
+  const { accountsApi } = useApi();
+  const { data: stxBalance } = useQuery(
```

Comment:
just an idea: pulling this query out into its own file would make it more reusable and help whoever might be needing it in the future to make sure they use the same query key

Reply:
I wanted to do this initially but thought maybe it's premature optimization as this query is only used here, no strong opinion. I also wanted to refactor query keys across the app as they're now scattered everywhere (some hardcoded, some defined multiple times in different constant files)